broker-core/src/main/java/io/zeebe/broker/workflow/processor/gateway/ExclusiveSplitHandler.java

```diff
@@ -43,16 +43,31 @@ public void handle(BpmnStepContext<ExecutableExclusiveGateway> context) {
 
       if (sequenceFlow != null) {
         value.setActivityId(sequenceFlow.getId());
-        context.getStreamWriter().writeNewEvent(WorkflowInstanceIntent.SEQUENCE_FLOW_TAKEN, value);
+        context.getOutput().writeNewEvent(WorkflowInstanceIntent.SEQUENCE_FLOW_TAKEN, value);
       } else {
         final String errorMessage = "All conditions evaluated to false and no default flow is set.";
-        context.raiseIncident(ErrorType.CONDITION_ERROR, errorMessage);
+        raiseIncident(context, errorMessage);
       }
     } catch (JsonConditionException e) {
-      context.raiseIncident(ErrorType.CONDITION_ERROR, e.getMessage());
+      raiseIncident(context, e.getMessage());
     }
   }
 
+  private void raiseIncident(
+      BpmnStepContext<ExecutableExclusiveGateway> context, final String errorMessage) {
+    context.raiseIncident(ErrorType.CONDITION_ERROR, errorMessage);
+
+    // TODO: this is a hack to avoid that we believe this token is consumed when the incident is
```

Comment:
This a problem when we decrement on event consuming and increment, if events are written right?
Is it possible that we only increment on scope creation and decrement on scope termination? I'm not sure if we already discussed that sorry :sweat_smile: Maybe we can discuss this f2f again.